src/squishy/core.clj

```diff
@@ -1,15 +1,27 @@
 (ns squishy.core
-  (:require [squishy.data-readers]
+  (:require [squishy.data-readers :as dr]
             [cemerick.bandalore :as sqs]
-            [clojure.tools.logging :as log]))
+            [clojure.tools.logging :as log])
+  (:import  [com.amazonaws.regions Region]))
 
 (def default-visibility-timeout 30) ; seconds
 
-(defn client [access-key secret-key region]
-  (doto (sqs/create-client access-key
-                           secret-key)
+(defmulti client
+  "Construct an SQS client instance from the provided credentials and region.
+   Region can be a string in either name (`us-east-1`) or enum (`US_EAST_1`)
+   format, or it can be an instance of com.amazonaws.regions.Region already."
+  (fn [_ _ region] (type region)))
+
+(defmethod client com.amazonaws.regions.Region [access-key secret-key region]
```

Comment:
Hmm, [Amazon SQS say `setRegion` is deprecated](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/sqs/AmazonSQS.html#setRegion-com.amazonaws.regions.Region-), but I don't (immediately) see a suggestion about _what to do instead_.

Reply:
Yeah, the approved way to do this now is to use the `AmazonSQSClientBuilder` because we need much ceremony in our code. Let me check to see if `bandalore` has a better story around this, but I suspect it doesn't.

Reply:
Yeah, `bandalore` hasn't been upgraded in 5 years. I suspect if we want to continue using and offering this library, we should upgrade it with new underlying AWS connection mechanisms.

Reply:
Would it make sense to open an issue and leave the deprecation for now?

Reply:
Yeah, that sounds like a reasonable next step. It's pretty rare in my experience for a deprecated thing to _actually_ go away, and this PR doesn't introduce new deprecations, just continues using one. I'll start an issue to upgrade this library to something more modern that doesn't have deprecated uses of the Java classes. 4.0.0 here we come!